File: python/quick.py
# time : O(n log n)
# Space: O(n)

from time import time
from random import randint

raw_data = []
def random_generator(target):
    for i in range(target):
        raw_data.append(randint(0, target))

def quick(data):
    n = len(data)

    if n < 1:
        return data
    else:
        left, right = [], []

        pivot = data[-1]
        for i in data[:-1]:
            if i < pivot:
                left.append(i)
            else:
                right.append(i)

        left = quick(left)
        right = quick(right)
        return left + [pivot] + right
        # return sorted_
random_generator(10)
print(f"Raw data: {raw_data}")
start = time()
sorted_ = quick(raw_data)
finish = time()
print(f"Sorted data: {sorted_}")
print(f"Time: {finish-start}")

File: javascript/quick.js
unsorted = [1,5,7,4,10,8,3,2,0,6];

function quick(array)  {
   n = array.length;
  if (n < 2) {
    return array;
  } else  {
    let left = [];
    let right = [];
    let pivot = array[n - 1];
    for (let i = 0; i < n - 1; i++) {
      if (array[i] < pivot) {
        left.push(array[i]);
      } else  {
        right.push(array[i]);
      }
    }
    
    return [...quick(left), pivot, quick(right)];
    
  }
}


console.log("Unsorted array: " + unsorted);
sorted = quick(unsorted);
console.log("Sorted array: " + sorted);